cmd/kyma/test/definitions/opts.go

```diff
@@ -2,10 +2,10 @@ package definitions
 
 import "github.com/kyma-project/cli/internal/cli"
 
-type options struct {
+type Options struct {
```

Comment:
Is there a reason to export all the `options` types in all test subcommands?

Reply:
yes

_Reason:_
The `NewOptions()` functions is used outside the pkg and if the `options` is not exported then callers of that function from other packages will struggle to use it. For instance, they can't put a *options field in their own struct. So this is the reason why the golint returns such error.

Reply:
What error dos the golint return?

Unexporting the options is intended like that so that no other commands can manipulate the specific options for the command they belong to.

Is there a scenario where a parent command needs to use the internal fields of this struct?

Reply:
> What error dos the golint return?

` exported func NewOptions returns unexported type *github.com/kyma-project/cli/cmd/kyma/test/status.options, which can be annoying to use (golint)
func NewOptions(o *cli.Options) *options {`

---
is not about using the internal fields of that struct is about using that struct at all. 

> Unexporting the options is intended like that so that no other commands can manipulate the specific options for the command they belong to.

it's not exactly true, when you unexport the `Options` struct then you are not allowed to assign that outside of that package even if you are exposing the public constructor. You still can expose the Options and have a private fields which are not accessible outside the package


pseudo-code:
```go
package main 

func main() {
# you can do sth like that when returned option is private
opts := definitions.NewOptions()

opts.Foo()
}
```


```go
package main

// you cannot have a field because options are private
type Bar struct {
	opts *options
}

func main() {

	obj := Bar{
		opts: definitions.NewOptions(),
	}

}
```

Reply:
Ah I see... so the problem is not being able to embed this type in others. Do we have a scenario where we need to do this?

Otherwise it is reasonable that we have these options only available on the command they are relevant for. Others should only be able to create an instance to pass it to the command when being created, but that's it.

Reply:
> Do we have a scenario where we need to do this?
Currently, is not a case because of that you code compiles...

Let me put  this in this way:
Current implementation **violate the go lint rule**

What you want to do?
a) silence this error 
b) by idiomatic and fix that problem

Why you do not want to export that? If you have a public constructor then you return also the public struct, simple as that. Why you do not want to do that? It's just the best practices. You know already the pros and cons what else we want to discuss?

Reply:
I am trying to come up with a good solution which also makes the linter happy.

I just try to find other ways because there is a general tendency when writing go code to export everything blindly.

One way I can come up with is just changing the subcommand constructor to take `cli.Options` as param and the unexported options creation and initialisation is completely happening inside the constructor.

Reply:
How about `// nolint`?